nupic/regions/CLAClassifierRegion.py

```diff
@@ -256,16 +340,36 @@ def customCompute(self, recordNum, patternNZ, classification):
                    4 : [0.2, 0.4, 0.3, 0.5]}
     """
 
-    return self._claClassifier.compute( recordNum=recordNum,
-                                        patternNZ=patternNZ,
-                                        classification=classification,
-                                        learn = self.learningMode,
-                                        infer = self.inferenceMode)
+    return self._claClassifier.compute(recordNum=recordNum,
+                                       patternNZ=patternNZ,
+                                       classification=classification,
+                                       learn=self.learningMode,
+                                       infer=self.inferenceMode)
 
+  def getOutputValues(self, outputName):
+    """
+    Return the dictionary of output values. Note that these are normal Python
+    lists, rather than numpy arrays. This is to support lists with mixed scalars
+    and strings, as in the case of records with categorical variables
+    """
+    return self._outputValues[outputName]
+
+  def getOutputElementCount(self, name):
+    """Returns the width of dataOut."""
 
+    if name == "categoriesOut":
+      return len(self.stepsList)
+    elif name == "probabilities":
+      return len(self.stepsList) * self.numCategories
+    elif name == "actualValues":
+      return self.numCategories
+    else:
+      raise _UnknownOutput("Unknown output {}.".format(name))
```

Comment:
I'd just use `ValueError` rather than defining a new exception type. The purpose of defining your own is to give fine grained ability to catch exceptions but `ValueError` is plenty specific. Describe the exception thrown in the doc string